src/cartridges/int_adyen_SFRA/cartridge/adyen/scripts/payments/adyenTerminalApi.js

```diff
@@ -102,33 +122,33 @@ function createTerminalPayment(order, paymentInstrument, terminalId) {
       let paymentResponse = '';
       if (terminalResponse.SaleToPOIResponse) {
         paymentResponse = terminalResponse.SaleToPOIResponse.PaymentResponse;
+        const pspReference = pspReferenceFromResponse(paymentResponse);
         if (paymentResponse.Response.Result === 'Success') {
+          const paymentReceipt = parsePaymentReceipt(paymentResponse);
+
+          // Set attributes for OMS
           order.custom.Adyen_eventCode = 'AUTHORISATION';
-          let pspReference = '';
-          if (
-            !empty(
-              paymentResponse.PaymentResult.PaymentAcquirerData
-                .AcquirerTransactionID.TransactionID,
-            )
-          ) {
-            pspReference =
-              paymentResponse.PaymentResult.PaymentAcquirerData
-                .AcquirerTransactionID.TransactionID;
-          } else if (
-            !empty(paymentResponse.POIData.POITransactionID.TransactionID)
-          ) {
-            pspReference =
-              paymentResponse.POIData.POITransactionID.TransactionID.split(
-                '.',
-              )[1];
-          }
-          // Save full response to transaction custom attribute
-          paymentInstrument.paymentTransaction.transactionID = pspReference;
           order.custom.Adyen_pspReference = pspReference;
-          order.setPaymentStatus(Order.PAYMENT_STATUS_PAID);
-          order.setExportStatus(Order.EXPORT_STATUS_READY);
+          order.custom.Adyen_paymentMethod = paymentReceipt.paymentMethod;
+          paymentInstrument.custom.adyenMainPaymentInstrument = paymentResponse?.PaymentResult?.PaymentInstrumentData?.PaymentInstrumentType;
+          paymentInstrument.custom.adyenPaymentMethod = paymentReceipt.paymentMethod;
+          paymentInstrument.custom.Adyen_Payment_Method_Variant = paymentReceipt.paymentMethodVariant;
+          paymentInstrument.custom[`${constants.OMS_NAMESPACE}__Adyen_Payment_Method`] = paymentReceipt.paymentMethod;
+          paymentInstrument.custom[`${constants.OMS_NAMESPACE}__Adyen_Payment_Method_Variant`] = paymentReceipt.paymentMethodVariant;
+          paymentInstrument.paymentTransaction.transactionID = pspReference;
+          paymentInstrument.paymentTransaction.custom.Adyen_pspReference = pspReference;
+          paymentInstrument.paymentTransaction.custom.Adyen_paymentMethod = paymentReceipt.paymentMethod;
+          paymentInstrument.paymentTransaction.custom.authCode = 'AUTHORISATION';
```

Comment:
Should we rely on API response for this event code instead of hardcoding it to 'Authorisation'?